### PR TITLE
Added HCT register

### PR DESCRIPTION
**Pull request by carapai**

### CODE REVIEW

**Comment by ssmusoke** [comment]
@carapai Add the register to the list of reports - as its difficult to find now

File: api/src/main/java/org/openmrs/module/ugandaemrreports/common/Observation.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
+package org.openmrs.module.ugandaemrreports.common;
+
+import java.util.Date;
+
+/**
+ * Created by carapai on 26/11/2017.
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/common/Observation.java]: @carapai Remove this created by

File: api/src/main/java/org/openmrs/module/ugandaemrreports/common/PatientEncounterObs.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
+package org.openmrs.module.ugandaemrreports.common;
+
+import com.google.common.collect.Multimap;
+
+import java.util.Date;
+import java.util.List;
+
+/**
+ * Created by carapai on 26/07/2017.
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/common/PatientEncounterObs.java]: @carapai Remove this created by

File: api/src/main/java/org/openmrs/module/ugandaemrreports/common/PatientEncounterObs.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+ */
+public class PatientEncounterObs {
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/common/PatientEncounterObs.java]: @carapai Describe what this class is for

File: api/src/main/java/org/openmrs/module/ugandaemrreports/common/Observation.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+ */
+public class Observation {
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/common/Observation.java]: @carapai What is this class for? Put a description at the top

File: api/src/main/java/org/openmrs/module/ugandaemrreports/reports/Helper.java [hunk 5]
```diff
@@ -352,7 +452,7 @@ public static String getObsPeriod(Date period, Enums.Period periodType) {
                 .build();
     }
 
-    private static Connection getDatabaseConnection(Properties props) throws ClassNotFoundException, SQLException {
+    public static Connection getDatabaseConnection(Properties props) throws ClassNotFoundException, SQLException {
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/reports/Helper.java]: @carapai This seems misplaced should it not be in a TestHelper class as its not used during running within OpenMRS environment

File: api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetupHCTRegister.java [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+import org.openmrs.module.ugandaemrreports.library.DataFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * Health Unit TB Register Report
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetupHCTRegister.java]: @carapai This is the HTC Register

File: api/src/main/java/org/openmrs/module/ugandaemrreports/reports/Helper.java (hunk 5 above)
Comment (carapai) [reply api/src/main/java/org/openmrs/module/ugandaemrreports/reports/Helper.java]: This is used, uses openmrs to get mysql details at runtime

**Pull request merged**